pycore/torchparse.py

```diff
@@ -0,0 +1,72 @@
+import sys
+
+from torchinfo import summary
+
+import pycore.tikzeng as pnn
+
+
+class TorchArchParser:
+
+    text_mapping = {
+        "Linear": "\\mathrm{{FC}}",
+        "ReLU": "\\varphi_\\mathrm{{ReLU}}"
+    }
+
+    def __init__(self, torch_module, input_size):
+
+        self.torch_module = torch_module
+        self.summary_list = summary(self.torch_module, input_size=input_size).summary_list
+
+        self.arch = self.parse(self.summary_list)
+
+    def get_arch(self):
+
+        return self.arch
+
+    @staticmethod
+    def parse(summary_list):
+
+        arch = list()
+        arch.append(pnn.to_head(".."))
+        arch.append(pnn.to_cor())
+        arch.append(pnn.to_begin())
+        for idx, layer in enumerate(summary_list[2:], start=1):
+
+            if layer.class_name == "Linear":
```

Comment:
If more module types would be parsed in the future, having helper functions or a builder class corresponding to the layer.class_name would omit cluttering the `parse()` function with further if conditions.

Reply:
I agree and this becomes even more important with increased types of supported PyTorch modules (layers, activations). Though, I have oriented myself on the coding style in `blocks.py` and `tikzeng.py`. 

Further, I propose to abstract parsing and constructing the tikz code toward a collection of layers that map to a similar tikz representation and a general activation representation that maps to all possible PyTorch-implemented activation functions.

Though, all of this could - and imo should - be an improvement and extension on-top of this basic functionaility.